dask/array/creation.py

```diff
@@ -679,3 +680,252 @@ def tile(A, reps):
+    elif (isinstance(pad_width, Sequence) and
+          all(isinstance(pw, Integral) for pw in pad_width) and
+          len(pad_width) == 1):
+        pad_width = array.ndim * ((pad_width[0], pad_width[0]),)
+    elif (isinstance(pad_width, Sequence) and
+          len(pad_width) == 2 and
+          all(isinstance(pw, Integral) for pw in pad_width)):
+            pad_width = tuple(
+                (pad_width[0], pad_width[1]) for _ in range(array.ndim)
+            )
+    elif (isinstance(pad_width, Sequence) and
+          len(pad_width) == array.ndim and
+          all(isinstance(pw, Sequence) for pw in pad_width) and
+          all((len(pw) == 2) for pw in pad_width) and
+          all(all(isinstance(w, Integral) for w in pw) for pw in pad_width)):
+            pad_width = tuple((pw[0], pw[1]) for pw in pad_width)
+    else:
+        raise TypeError(
+            "`pad_width` must be composed of integral typed values."
+        )
+
+    return pad_width
+
+
+def np_pad(array, pad_width, mode, extra_arg=None):
+    if mode in ["maximum", "mean", "median", "minimum"]:
+        extra_arg = extra_arg or None
+        return np.pad(array, pad_width, mode, stat_length=extra_arg)
+    elif mode == "constant":
+        extra_arg = extra_arg or 0
+        return np.pad(array, pad_width, mode, constant_values=extra_arg)
+    elif mode == "linear_ramp":
+        extra_arg = extra_arg or 0
+        return np.pad(array, pad_width, mode, end_values=extra_arg)
+    elif mode in ["reflect", "symmetric"]:
+        extra_arg = extra_arg or "even"
+        return np.pad(array, pad_width, mode, reflect_type=extra_arg)
+    else:
+        return np.pad(array, pad_width, mode)
+
+
+def pad_edge(array, pad_width, mode, *args):
+    """
+    Helper function for padding edges.
+
+    Handles the cases where the only the values on the edge are needed.
+    """
+
+    token = tokenize(array, pad_width, mode, args)
+    name = 'pad-' + token
+    numblocks = array.numblocks
+
+    chunks = list()
+    for d, c in enumerate(array.chunks):
+        c = list(c)
+        c[0] += pad_width[d][0]
+        c[-1] += pad_width[d][-1]
+        c = tuple(c)
+        chunks.append(c)
+    chunks = tuple(chunks)
+
+    dsk = {}
+    for idx in product(*(range(n) for n in numblocks)):
+        pad_chunk_width = []
+        for d, i in enumerate(idx):
+            ith_pad_chunk_width = [0, 0]
+            if i == 0:
+                ith_pad_chunk_width[0] = pad_width[d][0]
+            if i == numblocks[d] - 1:
+                ith_pad_chunk_width[1] = pad_width[d][1]
+
+            ith_pad_chunk_width = tuple(ith_pad_chunk_width)
+            pad_chunk_width.append(ith_pad_chunk_width)
+
+        pad_chunk_width = tuple(pad_chunk_width)
+
+        array_chunk_key = (array.name,) + idx
+        result_chunk_key = (name,) + idx
+
+        if any(map(any, pad_chunk_width)):
+            dsk[result_chunk_key] = (
+                np_pad, array_chunk_key, pad_chunk_width, mode
```

Comment:
Currently we reuse NumPy's `pad` here. This increases the chunks on the edge of the array and simplifies the logic we need to handle.

An alternative to this would be to do the padding ourselves manually in Dask. The benefit being the input array would be unchanged and the padded data would appear as new chunks on the exterior of the array.

All of the other padding cases already add the padding using new, distinct chunks while not changing the array itself. Could be useful if we wanted to provide a user option to override the chunking of the padded array.